rotkehlchen/exchanges/kraken.py

```diff
@@ -1146,7 +1062,17 @@ def history_event_from_kraken(
 
                 # Make sure to not generate an event for KFEES that is not of type FEE
                 if asset != A_KFEE:
-                    event = HistoryEvent(
+                    event = AssetMovement(
+                        timestamp=timestamp,
+                        location=Location.KRAKEN,
+                        event_type=event_type,  # type: ignore  # will be deposit or withdrawal
```

Comment:
Hmmm I think that using the `x = y if z else k` ternary operator usage here is confusing mypy and probably also the reader. Is it enforced by linting? Or can't you just do

```python
if event_type in {HistoryEventType.DEPOSIT,  HistoryEventType.WITHDRAWAL}:
    event = AssetMovement()
else:
    event = historyEvent()
```

Reply:
Changed it to use a normal `if` like this, but had to rename `event` to `history_event` several places so that mypy wouldn't think that `event` always had to be a HistoryEvent when that variable name is reused later in the function. Also simplified things further by using the `create_asset_movement_with_fee` function.